gflownet.py

```diff
@@ -638,30 +660,32 @@ def flowmatch_loss(self, it, batch):
                 seq_oracle = self.env.seq2oracle([seq])
                 output_proxy = self.env.proxy(seq_oracle)
                 reward = self.env.proxy2reward(output_proxy)
-                print(idx, output_proxy, reward)
                 import ipdb
 
                 ipdb.set_trace()
 
         # Q(s,a)
         parents_Qsa = self.model(parents)[torch.arange(parents.shape[0]), actions]
 
+
         # log(eps + exp(log(Q(s,a)))) : qsa
-        in_flow = torch.log(
+        in_flow = torch.log(self.loss_eps + 
```

Comment:
why?

Reply:
In the paper, an epsilon is added both in the inflow and the outflow before making the difference of the logs